compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -203,7 +203,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 body,
                 ..
             }) => {
-                self.with_new_scopes(ident.span, |this| {
+                self.with_new_scopes(span, |this| {
```

Comment:
I had to do this change as otherwise the span points to the identifier of functions instead of the `fn`, so adding `gen` would have resulted in `fn gen foo` instead of `gen fn foo`.